app/src/main/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragment.kt

```diff
@@ -22,23 +23,74 @@ class ConceptCardFragment : InjectableDialogFragment() {
 
   companion object {
     /** The fragment tag corresponding to the concept card dialog fragment. */
-    const val CONCEPT_CARD_DIALOG_FRAGMENT_TAG = "CONCEPT_CARD_FRAGMENT"
+    private const val CONCEPT_CARD_DIALOG_FRAGMENT_TAG = "CONCEPT_CARD_FRAGMENT"
 
-    /**
-     * Creates a new fragment to show a concept card.
-     *
-     * @param skillId the skill ID for which a concept card should be loaded
-     * @param profileId the profile in which the concept card will be shown
-     * @return a new [ConceptCardFragment] to display the specified concept card
-     */
-    fun newInstance(skillId: String, profileId: ProfileId): ConceptCardFragment {
+    private fun newInstance(skillId: String, profileId: ProfileId): ConceptCardFragment {
       return ConceptCardFragment().apply {
         arguments = Bundle().apply {
           putString(SKILL_ID_ARGUMENT_KEY, skillId)
           putProto(PROFILE_ID_ARGUMENT_KEY, profileId)
         }
       }
     }
+
+    /**
+     * Removes any [ConceptCardFragment] in the given FragmentManager that is not of the given
+     * skill Id. If no [ConceptCardFragment] remains, creates a new fragment to show a concept card
+     * for the given skill id.
+     *
+     * @param skillId the skill ID for which a concept card should be loaded.
+     * @param profileId the profile in which the concept card will be shown.
+     * @param fragmentManager the [FragmentManager] where to show the concept card.
+     * @return the fragment created or null if none.
+     */
+    fun bringToFrontOrCreateIfNew(
+      skillId: String,
+      profileId: ProfileId,
+      fragmentManager: FragmentManager
+    ): ConceptCardFragment? {
```

Comment:
Is the return actually useful? It seems like none of the callsites use it (unless I missed one of them).

Reply:
I use the return in tests.

Reply:
Having an API specifically for tests isn't generally preferred unless that API may be used by non-tests in the future, or is test-only (since it sort of skews the expected usage of the API between production and test contexts).

If the goal here is just to verify that the correct outcome happens, could we instead inspect the activity's fragment manager to see if the correct fragment has been created/not created rather than depending on the return value of this function? I think that will result in both more robust tests and a cleaner API.

Reply:
Done